### ISSUE TITLE
Context values are not available if cdk is not global

### ISSUE THREAD

**Issue by DamirAinullin** [aws/aws-cdk#7239]
Context values are not available if I use cdk from dev dependencies perfoming "npm run cdk deploy" command. But it works well if I use just "cdk deploy" from global NPM package. Is it bug or expected behavior?

### Reproduction Steps

During deploy:
```
npm run cdk deploy --context configFile='filename'
```
And in project I try to get it:
```
const configFile = app.node.tryGetContext('configFile');
```
configFile variable is undefined. But if I run:
```
cdk deploy --context configFile='filename'
```
configFile variable will contain 'filename' string as expected.

### Environment

  - **CLI Version      :1.31
  - **Framework Version:1.31
  - **OS               :Windows server 2016
  - **Language         :typescript

### Other

--- 

This is :bug: Bug Report

**Comment by DamirAinullin** [comment]
@SomayaB @shivlaks  Do you have any thoughts about this problem?

**Comment by patrobinson** [comment]
I think I'm experiencing the same issue. I'm trying to integrate CDK with my project so it can synthesize and deploy a cloudformation template itself. But the `Vpc.fromLookup` returns a dummy value. It seems like `tryGetContext` isn't returning the VPC ID, even though I am passing it in.

I tried looking through the code to find where `tryGetContext` gets executed but got lost in a maze of metaprogramming.

**Comment by shivlaks** [comment]
@DamirAinullin sorry, I missed this one. can you share what your config file looks like? I'll start with getting a little repro going to investigate. 

@patrobinson - are you calling `tryGetContext` in your code as well? can you run `cdk synth --v` to get the detail of why the context lookup failed? If you're also able to provide a minimal repro of your application/setup, I'll start digging.

The dummy value is a marker that a context value was not found and missing context was reported by a construct. It can happen if context is missing or an error occurred during context retrieval. It represents an invalid state and cannot be deployed.

**Comment by DamirAinullin** [comment]
@shivlaks configFile in my case is just a string value. It is just a parameter which I want to get in my CDK project. It can be anything else, param1 for example. You can reproduce the case with initial project.

**Comment by shivlaks** [comment]
@DamirAinullin have you tried running the command with `npx cdk deploy --context configFile='filename'`. you should not need a global install. 

`npx` will check whether the command exists in $PATH, or in the local project binaries, and if so it will execute it.

**Comment by DamirAinullin** [comment]
Lets keep it for awhile. I will try to reproduce the problem using npx.

**Comment by gbsmit03** [comment]
I think the issue is with how command arg flags have to be passed when running via npm. Try `cdk deploy -- --context configFile='filename'`, note the extra --

https://docs.npmjs.com/cli/run-script

**Comment by aripalo** [comment]
Yeah, I just stumbled upon this as well.

## Steps to reproduce:

This assumes `cdk` CLI **not** installed globally. Not sure if the effect is the same with global installation, didn't have time to verify.

1. Initialize a new CDK TypeScript application with CDK version `1.59.0`:
    ```sh
    mkdir -p /tmp/demo
    cd /tmp/demo
    npx cdk@1.59.0 init --language typescript
    ```
2. See `scripts` section in `package.json`:
    ```jsonc
    "scripts": {
        "build": "tsc",
        "watch": "tsc -w",
        "test": "jest",
        "cdk": "cdk" // NOTE that cdk is added in scripts section
    },
    ```
    Since `cdk` is added into npm `scripts`, it suggests that one should use it via `npm run cdk`.

3. Edit `lib/demo-stack.ts` by adding `tryGetContext`:
    ```ts
    import * as cdk from '@aws-cdk/core';
    
    export class DemoStack extends cdk.Stack {
      constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        // The code that defines your stack goes here
        new cdk.CfnOutput(this, "ValueFromRuntimeContext", {
          value: this.node.tryGetContext("foo"),
        });
      }
    }
    ```

4. Synthetize the stack via `npm run cdk`:
    ```sh
    $ npm run cdk synth --context foo=bar
    
    > demo@0.1.0 cdk /private/tmp/demo
    > cdk "synth" "foo=bar"
    
    Missing value for CloudFormation output at path "DemoStack/ValueFromRuntimeContext"
    Subprocess exited with error 1
    npm ERR! code ELIFECYCLE
    npm ERR! errno 1
    npm ERR! demo@0.1.0 cdk: `cdk "synth" "foo=bar"`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the demo@0.1.0 cdk script.
    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /Users/******/.npm/_logs/2020-08-18T15_35_06_713Z-debug.log
    ```

5. Let's try again with `npx cdk`:
    ```sh
    $ npx cdk synth --context foo=bar
    Outputs:
      ValueFromRuntimeContext:
        Value: bar
    ```

## Solutions

I guess it should be better to leave `cdk` out of npm `scripts` section and just advice users to use `npx cdk`, since it should work no matter if you have cdk installed globally or not.

**Comment by gbsmit03** [comment]
@aripalo have you tried running the command like this `npm run cdk synth -- --context foo=bar`? Note the extra `--`? From the [NPM docs on running scripts](https://docs.npmjs.com/cli/run-script), you have to use "The special option --" to ensure the arguments are passed to your specified script. 

I resolved this same issue on my end by doing this. I'm running CDK with the projects locally installed modules.

**Comment by aripalo** [comment]
@gbsmit03 Yeah that works as well, but I use the `npx` command now as a default, but just wanted to document the steps how to reproduce the issue. Might be something that CDK peeps might want to look at some point, because it can confuse newcomers.

**Comment by joekiller** [comment]
After https://github.com/aws/aws-cdk/issues/7239#issuecomment-675555860 this hit me and I had to go back to node16 builds
